packages/astro/src/integrations/index.ts

```diff
@@ -34,6 +34,9 @@ export async function runHookConfigSetup({
 				updateConfig: (newConfig) => {
 					updatedConfig = mergeConfig(updatedConfig, newConfig) as AstroConfig;
 				},
+				injectRoute: (injectRoute) => {
+					updatedConfig._ctx.injectedRoutes = [...(updatedConfig._ctx.injectedRoutes || []), injectRoute];
```

Comment:
Does this need to be immutable for some reason? Can't we just use `.push(injectRoute)`?

Reply:
Ah, no this can be `.push` now. In an earlier commit there was no default value for `._ctx.injectedRoutes`, so I had to guard, but now it defaults to an empty array